javascript/javascript2/week2/Jsweek2.js

```diff
@@ -0,0 +1,183 @@
+ // Flight booking fullname function
+ 
+ 
+  let  fullname1 = ["Benjamin", "Hughes", true,"male"];
+  let fullname2 = ["Lukas", "Andrias",false,"male"]; 
+  let fullname3 = ["Sofia", "Larsen",true,"female"]; 
+   function getFullname (firstname, surname) {  
+    if (firstname=="" || surname=="" || typeof firstname !== "string" ||
+         typeof surname !== "string"){
+    return `Please enter a valid name.`
+       }
+    return  ` ${firstname} ${surname}` ; 
+ }
+ console.log(getFullname(fullname1[0], fullname1[1]));
+ console.log(getFullname(fullname2[0], fullname2[1])); 
+    
+ // Formal fullname 
+ 
+ 
+   function getFullname (firstname, surname, useFormalName) { 
+    if (firstname=="" || surname=="" || typeof firstname !== "string" ||
+         typeof surname !== "string"){
+    return `Please enter a valid name.`
+       }
+     else if (useFormalName) { 
+     return `Lord ${firstname} ${surname}` ; 
+  }
+  else 
+  return  `${firstname} ${surname}` ; 
+ };
+ console.log(getFullname(fullname1[0], fullname1[1],fullname1[2]));
+ console.log(getFullname(fullname2[0], fullname2[1],fullname2[2]));  
+ 
+ //  Formal fullname gender specific 
+ 
+   function getFullname (firstname, surname, useFormalName,gender) { 
+     if (firstname=="" || surname=="" || typeof firstname !== "string" ||
+         typeof surname !== "string"){
+    return `Please enter a valid name.`
+     }
+    else if (gender=="female" && useFormalName){
```

Comment:
you could also check first `if (useFormalName)` and inside if check `gender`
```suggestion
    else if (useFormalName){
                 if (gender === "female") {
```